app/models/translation_override.rb

```diff
@@ -105,6 +105,22 @@ def self.expire_cache(locale, key)
     true
   end
 
+  def self.transform_pluralized_key(key)
+    match = key.match(/(.*)\.(zero|two|few|many)\z/)
+    match ? match.to_a.second + ".other" : key
+  end
+
+  def self.custom_interpolation_keys(key)
+    transformed_key = transform_pluralized_key(key)
+
+    custom_keys =
+      ALLOWED_CUSTOM_INTERPOLATION_KEYS.find do |keys, value|
+        break value if keys.any? { |key| transformed_key.start_with?(key) }
+      end
+
+    custom_keys || []
```

Comment:
Is the call to `transform_pluralized_key` needed? Wouldn't the following work as well? :thinking: 

```suggestion
    custom_keys =
      ALLOWED_CUSTOM_INTERPOLATION_KEYS.find do |keys, value|
        break value if keys.any? { |k| key.start_with?(k) }
      end

    custom_keys || []
```

Reply:
I'm also not sure. This is based on the `#check_interpolation_keys` method. If we change this tests are failing, so I'm a bit hesitant to just ✂️ it. It seems to have something to do with certain locales (e.g. Arabic) with custom plural forms.

Reply:
After looking over this again, the reason for this is we treat English as the source of truth for interpolation keys, but other languages, as mentioned, might have plural forms (e.g. zero, two, few, many) which aren't present in English, which has only one or other, so we need to map this before looking for the keys. Will add some comments as to why we do this, and double check that this is working correctly with regards to this feature. 🙏

Reply:
Thanks to this comment I found another bug where the keys didn't show when editing one of the irregular pluralization forms. This is now fixed. ❤️ 

<img width="233" alt="Screenshot 2023-06-28 at 12 57 08 PM" src="https://github.com/discourse/discourse/assets/5259935/08a7c209-1382-4fcd-86fd-734f3c554bfc">